### PR TITLE
feat: add new resources

### PR DESCRIPTION
**Pull request by frezbo**
feat: add new resources
    
Adds new resources:
- `talos_client_configuration`
- `talos_machine_configuration_apply`
- `talos_machine_bootstrap`
- `talos_cluster_kubeconfig`
    
Add new data sources:    
- `talos_cluster_kubeconfig`

Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -22,8 +29,46 @@ resource "talos_machine_configuration_worker" "machineconfig_worker" {
   machine_secrets  = talos_machine_secrets.machine_secrets.machine_secrets
 }
 
+resource "talos_client_configuration" "talosconfig" {
+  cluster_name    = var.cluster_name
+  machine_secrets = talos_machine_secrets.machine_secrets.machine_secrets
+  endpoints       = ["10.5.0.2"]
+}
+
+// use data source if you want a get a new talos client config each time TF is run
```
Comment (frezbo) [inline examples/basic/main.tf:38]: not sure if we should just use resources only or support both `data` and `resource`. Data source means a new client config is generated on each run

**Review by andrewrynhard — APPROVED** [review]
A lot of changes so possible I missed something. Will test for my use case.

**Comment by frezbo** [comment]
> A lot of changes so possible I missed something. Will test for my use case.

yes, I'm conflicted on what to do with the data sources since it's gets refreshed on each plan/apply. We could keep both or completely get away with data sources.

Yes, please do test and will get it merged then. I'll also test on the pulumi generated from this

File: examples/basic/main.tf (hunk 1 above)
Comment (frezbo) [reply examples/basic/main.tf:38]: plan to deprecate the data sources at a later point

**Pull request merged**